cmd/snap-recovery/recover/recover.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package recover
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/cmd/snap-recovery/partition"
+	"github.com/snapcore/snapd/gadget"
+)
+
+type Options struct {
+	// will contain encryption later
+}
+
+type Recover struct {
```

Comment:
Package as well as type-level descriptions would be appreciated.

Reply:
Yeah, this is all very early though, given the current structure of the Recover struct this may just become a single function and the package may actually be removed and just folded into the cmd/snap-recovery/ code. Sorry that its not quite so clean as usually, this is merging bit of Claudio and my own ideas which makes this a bit less straightforward than usual.